plugins/replay/src/main/java/forklift/replay/ReplayESWriter.java

```diff
@@ -36,35 +36,54 @@ public ReplayESWriter(String hostname, int port, String clusterName) {
 
     @Override
     protected void poll(ReplayESWriterMsg t) {
-        final String index = "forklift-replay-" + LocalDate.now().format(DateTimeFormatter.BASIC_ISO_DATE);
+        final String replayVersion = t.getFields().get("forklift-replay-version");
+        if ("3".equals(replayVersion)) { // latest version
+            String indexDate = t.getFields().get("first-processed-date");
+            if (indexDate == null)
+                indexDate = LocalDate.now().format(DateTimeFormatter.BASIC_ISO_DATE);
+            final String index = "forklift-replay-" + indexDate;
 
-        // In order to ensure there is only one replay msg for a given id we have to clean the msg id from
-        // any previously created indexes.
-        try {
-            final SearchResponse resp = client.prepareSearch("forklift-replay*").setTypes("log")
-                .setQuery(QueryBuilders.termQuery("_id", t.getId()))
-                .setFrom(0).setSize(50).setExplain(false)
-                .execute()
-                .actionGet();
+            try {
+                // Index the new information.
+                client.prepareIndex(index, "log")
+                    .setVersion(t.getVersion()).setVersionType(VersionType.EXTERNAL_GTE)
+                    .setId(t.getId()).setSource(t.getFields()).execute().actionGet();
+            } catch (VersionConflictEngineException expected) {
+                log.debug("Newer replay message already exists", expected);
+            }
+        } else if ("unknown".equals(replayVersion)) { // manage old versions, for backward compatibility
```

Comment:
Any chance that this could be `null`? Should probably handle it with an else case JIC?